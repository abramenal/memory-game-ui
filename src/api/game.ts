import { API_BASE_URL } from '../constants';

import { Game } from '../types';

type CreateGameParams = {
  userId: string;
  turnsAmount: number;
};

export const createGame = ({ userId, turnsAmount }: CreateGameParams): Promise<Game> => fetch(
  `${API_BASE_URL}/games`,
  {
    method: 'POST',
    body: JSON.stringify({
      userId,
      turnsAmount,
    }),
  },
).then((res) => res.json());

type SubmitGameParams = {
  gameId: string;
  userId: string;
  value: string;
};

export const submitGameTurn = ({ gameId, userId, value }: SubmitGameParams): Promise<Game> => fetch(
  `${API_BASE_URL}/games/${gameId}/turn`,
  {
    method: 'POST',
    body: JSON.stringify({
      userId,
      value,
    }),
  },
).then((res) => res.json());
