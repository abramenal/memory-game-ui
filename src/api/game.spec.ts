import { createGame, submitGameTurn } from './game';

describe('createGame', () => {
  beforeAll(() => {
    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
    // @ts-ignore
    globalThis.fetch = jest.fn((...args) =>
      Promise.resolve({
        json: () => args,
      }),
    );
  });

  it('calls HTTP API', async () => {
    const request = await createGame({ userId: 'test', turnsAmount: 4 });
    expect(request).toMatchSnapshot();
  });
});

describe('submitGameTurn', () => {
  beforeAll(() => {
    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
    // @ts-ignore
    globalThis.fetch = jest.fn((...args) =>
      Promise.resolve({
        json: () => args,
      }),
    );
  });

  it('calls HTTP API', async () => {
    const request = await submitGameTurn({ userId: 'test', gameId: 'test2', value: 4 });
    expect(request).toMatchSnapshot();
  });
});
