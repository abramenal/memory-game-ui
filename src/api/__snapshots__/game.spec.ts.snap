// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createGame calls HTTP API 1`] = `
Array [
  "http://localhost:3000/games",
  Object {
    "body": "{\\"userId\\":\\"test\\",\\"turnsAmount\\":4}",
    "headers": Object {
      "Content-Type": "application/json",
    },
    "method": "POST",
    "mode": "cors",
  },
]
`;

exports[`submitGameTurn calls HTTP API 1`] = `
Array [
  "http://localhost:3000/games/test2/turn",
  Object {
    "body": "{\\"userId\\":\\"test\\",\\"value\\":4}",
    "headers": Object {
      "Content-Type": "application/json",
    },
    "method": "POST",
    "mode": "cors",
  },
]
`;
